jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -23,25 +23,33 @@ class DeploymentDirectDeploy extends React.Component {
     super(props);
     this.state = {
       entityModel: false,
-      isBundle: this.props.ddData.id.indexOf('bundle') !== -1
+      isBundle: this.props.ddData.id.indexOf('bundle') !== -1,
+      loading: false
     };
   }
 
-  componentDidMount() {
-    this.props.getEntity(this.props.ddData.id, (error, data) => {
-      if (error) {
-        console.error('cannot fetch the entity:' + error);
-        return;
-      }
-      if (data.length > 0) {
-        data = data[0];
-        this.setState({
-          entityModel: this.props.makeEntityModel(data)
+  componentWillMount() {
+    this.setState({loading: true}, () => {
+      this.props.getEntity(this.props.ddData.id, (error, data) => {
+        this.setState({loading: false}, () => {
+          if (error) {
+            console.error('cannot fetch the entity:' + error);
+            return;
+          }
+          if (data && data.length > 0) {
```

Comment:
I'm not sure, this condition was existing, the case where there is no data is handle by: https://github.com/juju/juju-gui/pull/3074/files/83143dd5d810750ed1945c6efac217ef15c86406#diff-3422c41f09986a4e510694a8e2b309e3R115